llvmlite/tests/test_ir.py

```diff
@@ -323,6 +323,17 @@ def test_unnamed_metadata_3(self):
         self.assert_ir_line('!1 = !{ i32 789 }', mod)
         self.assert_ir_line('!2 = !{ i32 123, !0, !1, !0 }', mod)
 
+    def test_self_relf_metadata(self):
+        mod = self.module()
+        value = mod.add_metadata((), self_ref=True)
+        self.assert_ir_line('!0 = !{ !0 }', mod)
+        value2 = mod.add_metadata((value,))
+        assert value is not value2
+        self.assert_ir_line('!1 = !{ !0 }', mod)
+        value3 = mod.add_metadata((), self_ref=True)
+        self.assert_ir_line('!2 = !{ !2 }', mod)
+        assert value is not value3
```

Comment:
Can you add test for metadata that has other operands?

Reply:
done